### ISSUE TITLE
[m1b/infra] Experiment tracking (aim) integrated with Wittgenstein manifest spine

### ISSUE THREAD

**Issue by Jah-yee** [p-to-q/wittgenstein#399]
## Context

Per [`docs/research/2026-05-13-wittgenstein-research-program.md`](../blob/main/docs/research/2026-05-13-wittgenstein-research-program.md) §"Track 1" + "Phase 1 — Eval program", every training run must:

1. Get an experiment-tracker entry (loss curves, metric snapshots, run config).
2. Get a Wittgenstein training manifest carrying `experiment.uri` pointing at the tracker entry.

So a future reader can navigate from a release artifact -> its training manifest -> the tracker dashboard, fully reproducible.

This issue tracks setting that integration up before the three training slices ([#396](https://github.com/p-to-q/wittgenstein/issues/396) / [#397](https://github.com/p-to-q/wittgenstein/issues/397) / [#398](https://github.com/p-to-q/wittgenstein/issues/398)) start. It should be reviewed alongside the model-owner hub in #435 because tracker choice and manifest shape affect every training receipt.

### Pick the tracker

Default: **[`aim`](https://github.com/aimhubio/aim)** (Apache-2.0, self-hostable, no vendor lock-in). Reasons:
- Open-source, our license posture (ADR-0020) is friendly to permissive-only infra.
- Self-hostable: aim's web UI can run alongside the training cluster; no external SaaS dependency.
- Strong Pythonic API; integrates with PyTorch trivially.

Maintainer-overridable to W&B if there's a strong reason. MLFlow is a third option.

### Wire up the manifest spine

Extend `RunManifestSchema` with an optional `experiment` block:

```ts
experiment: z.object({
  tracker: z.enum(["aim", "wandb", "mlflow"]),
  uri: z.string().url(),            // direct link to the run's dashboard entry
  runId: z.string(),                 // tracker-side run ID
  hash: z.string().optional(),       // git SHA of the training-code repo (may differ from harness SHA)
}).optional(),
```

Training scripts under `research/training/` write this field on the manifest emitted at each checkpoint or release.

### Stand up the aim instance

Either run aim locally during this phase (`aim up`, port 43800) or behind a small VPS reverse proxy if multiple contributors share. Document the deploy in `docs/training/experiment-tracking.md`.

## Acceptance

- `aim` instance running, accessible to contributors.
- `RunManifestSchema.experiment` field landed and parsing tests added.
- Training script template under `research/training/_template/` that:
  - Initializes an aim run at start.
  - Logs scalars + hparams every step.
  - Emits a Wittgenstein training manifest at every checkpoint.
- `docs/training/experiment-tracking.md` describing the deploy + usage.
- #435 has an explicit owner-review note for the final tracker choice if the implementation does not stay with aim.

## Non-goals

- Not adding a Wittgenstein-shaped tracker. Aim covers our needs.
- Not standing up enterprise SSO / multi-tenant for the tracker; single shared instance is fine for the team size.

## Dependencies

- None. Can land before any training slice.
- Coordinate with #435 for model-owner review of tracker choice, manifest fields, and receipt expectations.

## Refs

- Research program: [§Track 1 + §Phase 1 Eval program](../blob/main/docs/research/2026-05-13-wittgenstein-research-program.md)
- Existing manifest spine: [`packages/schemas/src/manifest.ts`](../blob/main/packages/schemas/src/manifest.ts)
- Aim project: [aimhubio/aim](https://github.com/aimhubio/aim)

**Comment by koriyoshi2041** [comment]
Clarifying the reopen after #566:

- #566 landed the repository-side experiment tracking floor: the training experiment receipt schema, local JSONL tracker, shared/tokenizer smoke receipt emission, and docs that keep tracker linkage as a sibling receipt rather than a freeform TrainingRunManifest field.
- The original #399 acceptance still includes repo-external evidence: a shared Aim instance accessible to contributors, lab endpoint/credential/retention docs, and a contributor-visible deployed run.
- Keep this issue open until that lab/ops evidence exists. #566 should be treated as concrete repo plumbing progress, not full acceptance closure.
